mdal/mdal.cpp

```diff
@@ -201,105 +219,210 @@ void MDAL_M_CloseDataset( DatasetH dataset )
+    return nullptr;
+  }
+  MDAL::DatasetGroup *g = static_cast< MDAL::DatasetGroup * >( group );
+  return static_cast< DatasetH >( g->maximumDataset.get() );
+}
+
+int MDAL_G_metadataCount( DatasetGroupH group )
+{
+  if ( !group )
   {
     sLastStatus = MDAL_Status::Err_IncompatibleDataset;
     return 0;
   }
-  MDAL::Dataset *d = static_cast< MDAL::Dataset * >( dataset );
-  int len = static_cast<int>( d->metadata.size() );
+  MDAL::DatasetGroup *g = static_cast< MDAL::DatasetGroup * >( group );
+  int len = static_cast<int>( g->metadata.size() );
   return len;
 }
 
-const char *MDAL_D_metadataKey( DatasetH dataset, int index )
+const char *MDAL_G_metadataKey( DatasetGroupH group, int index )
 {
-  if ( !dataset )
+  if ( !group )
+  {
+    sLastStatus = MDAL_Status::Err_IncompatibleDataset;
+    return EMPTY_STR;
+  }
+  MDAL::DatasetGroup *g = static_cast< MDAL::DatasetGroup * >( group );
+  int len = static_cast<int>( g->metadata.size() );
+  if ( len <= index )
   {
     sLastStatus = MDAL_Status::Err_IncompatibleDataset;
     return EMPTY_STR;
   }
+  size_t i = static_cast<size_t>( index );
+  return _return_str( g->metadata[i].first );
+}
 
-  MDAL::Dataset *d = static_cast< MDAL::Dataset * >( dataset );
-  int len = static_cast<int>( d->metadata.size() );
+const char *MDAL_G_metadataValue( DatasetGroupH group, int index )
+{
+  if ( !group )
+  {
+    sLastStatus = MDAL_Status::Err_IncompatibleDataset;
+    return EMPTY_STR;
+  }
+  MDAL::DatasetGroup *g = static_cast< MDAL::DatasetGroup * >( group );
+  int len = static_cast<int>( g->metadata.size() );
   if ( len <= index )
   {
     sLastStatus = MDAL_Status::Err_IncompatibleDataset;
-    return nullptr;
+    return EMPTY_STR;
   }
   size_t i = static_cast<size_t>( index );
-  return d->metadata[i].first.c_str();
+  return _return_str( g->metadata[i].second );
 }
 
-const char *MDAL_D_metadataValue( DatasetH dataset, int index )
+const char *MDAL_G_name( DatasetGroupH group )
 {
-  if ( !dataset )
+  if ( !group )
   {
     sLastStatus = MDAL_Status::Err_IncompatibleDataset;
     return EMPTY_STR;
   }
+  MDAL::DatasetGroup *g = static_cast< MDAL::DatasetGroup * >( group );
+  return _return_str( g->name() );
+}
+
+bool MDAL_G_hasScalarData( DatasetGroupH group )
+{
+  if ( !group )
+  {
+    sLastStatus = MDAL_Status::Err_IncompatibleDataset;
+    return true;
+  }
+  MDAL::DatasetGroup *g = static_cast< MDAL::DatasetGroup * >( group );
+  return g->isScalar;
+}
+
+bool MDAL_G_isOnVertices( DatasetGroupH group )
+{
+  if ( !group )
+  {
+    sLastStatus = MDAL_Status::Err_IncompatibleDataset;
+    return true;
+  }
+  MDAL::DatasetGroup *g = static_cast< MDAL::DatasetGroup * >( group );
+  return g->isOnVertices;
+}
+
+///////////////////////////////////////////////////////////////////////////////////////
+/// DATASETS
+///////////////////////////////////////////////////////////////////////////////////////
+
+bool MDAL_D_hasScalarData( DatasetH dataset )
```

Comment:
Remove from API now that it is handled within groups? (same for MDAL_D_isOnVertices)